lang/java/avro/src/main/java/org/apache/avro/Conversions.java

```diff
@@ -68,6 +69,22 @@ public UUID fromCharSequence(CharSequence value, Schema schema, LogicalType type
     public CharSequence toCharSequence(UUID value, Schema schema, LogicalType type) {
       return value.toString();
     }
+
+    @Override
+    public UUID fromFixed(final GenericFixed value, final Schema schema, final LogicalType type) {
+      ByteBuffer buffer = ByteBuffer.wrap(value.bytes());
+      long mostSigBits = buffer.getLong();
+      long leastSigBits = buffer.getLong();
+      return new UUID(mostSigBits, leastSigBits);
+    }
+
+    @Override
+    public GenericFixed toFixed(final UUID value, final Schema schema, final LogicalType type) {
+      ByteBuffer buffer = ByteBuffer.allocate(2 * Long.BYTES);
```

Comment:
Do we want to re-use the buffer? This avoids reallocating the buffer each time.

https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/avro/ValueReaders.java#L396-L402

Reply:
```suggestion
      ByteBuffer buffer = ByteBuffer.allocate(2 * Long.BYTES);
      buffer.order(ByteOrder.BIG_ENDIAN);
```
I still think we cannot leave the endianness undefined. 

![image](https://github.com/apache/avro/assets/1134248/20a2b165-c137-4060-ad57-0e3f03bbc293)

According to Wikipedia, the big-endian format is the most common one (and the other one looks awkward :)

Reply:
Do we want to explicitly quote the spec here, and specify network byte order (i.e., big-endian)? In #2672 the Avro spec will explicitly mention RFC-4122, which defines the exact byte layout in [section 4.1.2](https://www.ietf.org/rfc/rfc4122.html#section-4.1.2).

Reply:
Also, a [`ByteBuffer`](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html) (as per its Javadoc) is initially always big-endian.

Reply:
I'm not fully comprehending the section 4.1.2. It is meant to clarify the byte order, but it doesn't help me a a lot:

> The fields are presented with the most significant one first.

That translates to big-endianness to me.

Reply:
Correct. This is explained a bit later in the section (the next page in the print version), where it says:
> In the absence of explicit application or presentation protocol specification to the contrary, [...] the fields are encoded as 16 octets, with the sizes and order of the fields defined above, and with each field encoded with the Most Significant Byte first (known as network byte order).

In other words: unless we explicitly want to document otherwise, everything is in network order (also known as big-endian).